linter.py

```diff
@@ -46,50 +59,95 @@ def cmd(self):
         command = ['flow']
         merged_settings = self.get_merged_settings()
 
+        command.extend(['check-contents', '@'])
+
         if merged_settings['show-all-errors']:
             command.append('--show-all-errors')
 
         command.append('--json')  # need this for simpler error handling
 
-        return self.build_cmd(command)
+        c = self.build_cmd(command)
+        persist.debug('flow attempting to run from: {}'.format(c))
+        return c
 
     def _error_to_tuple(self, error):
         """
         Map an array of flow error messages to a fake regex match tuple.
 
-        flow returns errors like this: {message: [{<msg>},..,]} where
-        <msg>:Object {
-            descr: str,
-            level: str,
-            path: str,
-            line: number,
-            endline: number,
-            start: number,
-            end: number
+        this is described in `flow/tsrc/flowResult.js` in the flow repo
+
+        flow returns errors like this
+        type FlowError = {
+            message: Array<FlowMessage>,
+            operation?: FlowMessage
+        }
+        type FlowMessage = {
+            descr: string,
+            type: "Blame" | "Comment",
+            context?: ?string,
+            loc?: ?FlowLoc,
+            indent?: number,
         }
 
         Which means we can mostly avoid dealing with regex parsing since the
         flow devs have already done that for us. Thanks flow devs!
         """
         error_messages = error.get('message', [])
-        match = self.filename == error_messages[0]['path']
         # TODO(nsfmc): `line_col_base` won't work b/c we avoid `split_match`'s codepath
-        line = error_messages[0]['line'] - 1
-        col = error_messages[0]['start'] - 1
-
-        level = error_messages[0].get('level', False) or error.get('level', '')
+        operation = error.get('operation', {})
+        loc = operation.get('loc') or error_messages[0].get('loc', {})
+
+        error_context = operation.get('context') or error_messages[0].get('context', '')
+        match = self.filename == loc.get('source')
+        message_start = loc.get('start', {})
+        message_end = loc.get('end', {})
+
+        line = message_start.get('line', None)
+        if line:
+            line -= 1
+        col = message_start.get('column', None)
+        if col:
+            col -= 1
+        end = message_end.get('column', None)
+
+        # slice the error message from the context and loc positions
+        if end and line == (message_end.get('line') - 1):
```

Comment:
for multiline errors, i punt and don't even try to match the text, instead, i let sublimelinter highlight the first character of the error and hope that the context messages are useful enough (they're usually ok)